src/back/wgsl/writer.rs

```diff
@@ -1916,14 +1924,25 @@ const fn image_dimension_str(dim: crate::ImageDimension) -> &'static str {
     }
 }
 
-const fn scalar_kind_str(kind: crate::ScalarKind) -> &'static str {
+const fn scalar_kind_str(kind: crate::ScalarKind, width: u8) -> &'static str {
     use crate::ScalarKind as Sk;
 
-    match kind {
-        Sk::Float => "f32",
-        Sk::Sint => "i32",
-        Sk::Uint => "u32",
-        Sk::Bool => "bool",
+    match (kind, width) {
+        // Floating point
+        (Sk::Float, 8) => "f64",
+        (Sk::Float, _) => "f32",
+        // Signed integer
+        (Sk::Sint, 1) => "i8",
+        (Sk::Sint, 2) => "i16",
+        (Sk::Sint, 8) => "i64",
+        (Sk::Sint, _) => "i32",
+        // Unsigned integer
+        (Sk::Uint, 1) => "u8",
+        (Sk::Uint, 2) => "u16",
+        (Sk::Uint, 8) => "u64",
+        (Sk::Uint, _) => "u32",
+        // Boolean
+        (Sk::Bool, _) => "bool",
```

Comment:
```suggestion
    match (kind, width) {
        (Sk::Float, 8) => "f64",
        (Sk::Float, 4) => "f32",
        (Sk::Sint, 4) => "i32",
        (Sk::Uint, 4) => "u32",
        (Sk::Bool, 1) => "bool",
        _ => unreachable!(),
```

We are removing the non-32bit integers in https://github.com/gfx-rs/naga/pull/2146 since the validator never allows those to go through.